exec/vector/src/main/java/org/apache/drill/exec/vector/accessor/impl/TupleReaderImpl.java

```diff
@@ -101,8 +88,61 @@ public String getAsString(int colIndex) {
       return "\"" + colReader.getString() + "\"";
     case DECIMAL:
       return colReader.getDecimal().toPlainString();
+    case ARRAY:
+      return getArrayAsString(colReader.array());
     default:
       throw new IllegalArgumentException("Unsupported type " + colReader.valueType());
     }
   }
+
+  private String bytesToString(byte[] value) {
+    StringBuilder buf = new StringBuilder()
+        .append("[");
+    int len = Math.min(value.length, 20);
+    for (int i = 0; i < len;  i++) {
+      if (i > 0) {
+        buf.append(", ");
+      }
+      buf.append((int) value[i]);
+    }
+    if (value.length > len) {
+      buf.append("...");
+    }
+    buf.append("]");
+    return buf.toString();
+  }
+
+  private String getArrayAsString(ArrayReader array) {
+    StringBuilder buf = new StringBuilder();
+    buf.append("[");
+    for (int i = 0; i < array.size(); i++) {
+      if (i > 0) {
+        buf.append( ", " );
+      }
+      switch (array.valueType()) {
```

Comment:
for valueType `MAP` and `ARRAY` we should `throw UnsupportedOperationException()`